src/wp-includes/rest-api.php

```diff
@@ -44,6 +44,12 @@ function register_rest_route( $namespace, $route, $args = array(), $override = f
 		return false;
 	}
 
+	$clean_namespace = trim( $namespace, '/' );
+
+	if ( $clean_namespace !== $namespace ) {
+		_doing_it_wrong( __FUNCTION__, __( 'Namespace must not start or begin with a slash.' ), '5.4.2' );
```

Comment:
> start or begin

Should it be `start or end`?

Reply:
That would make a lot more sense 😃. Thanks for catching that @todeveni! Fixed in e24bec1.